api/location/schema.py

```diff
@@ -154,6 +155,8 @@ class Query(graphene.ObjectType):
     def resolve_all_locations(self, info):
         query = Location.get_query(info)
         result = query.filter(LocationModel.state == "active")
+        for location in result:
+            location.rooms = filter_active_rooms(location.rooms)
```

Comment:
@mnswaleh Since there is a relationship between locations and rooms, I think you can use a join query to get all locations with their active rooms.